Add Document Image Transformer (DiT) (#15984)
huggingface/transformers: docs/source/model_doc/dit.mdx

document image classification, table detection and document layout analysis.

The abstract from the paper is the following:

*Image Transformer has recently achieved significant progress for natural image understanding, either using supervised (ViT, DeiT, etc.) or self-supervised (BEiT, MAE, etc.) pre-training techniques. In this paper, we propose DiT, a self-supervised pre-trained Document Image Transformer model using large-scale unlabeled text images for Document AI tasks, which is essential since no supervised counterparts ever exist due to the lack of human labeled document images. We leverage DiT as the backbone network in a variety of vision-based Document AI tasks, including document image classification, document layout analysis, as well as table detection. Experiment results have illustrated that the self-supervised pre-trained DiT model achieves new state-of-the-art results on these downstream tasks, e.g. document image classification (91.11 → 92.69), document layout analysis (91.0 → 94.9) and table detection (94.23 → 96.55). *

> One can directly plug in the weights of DiT into a BEiT model, like so:

Reviewer: There should be a link to the BEiT documentation as well so that inputs/classes available/tips can be seen easily

Author: I actually link to the BEiT docs twice, see sentence below

Reviewer 2: I would not mention BEiT here and change this sentence to:
```suggestion
One can directly use the weights of DiT with the AutoModel API:
```
and then talk about the fact it's BEiT behind the scenes at the end like you did.
